apps/builder/app/dashboard/projects/thumbnail-link.tsx

```diff
@@ -30,10 +32,15 @@ const getThumbnailAbbreviation = (title: string) =>
 export const ThumbnailLink = forwardRef<
   HTMLAnchorElement,
   { title: string; to: string }
->(({ title, to }, ref) => (
-  <Link ref={ref} to={to} className={thumbnailStyle()} tabIndex={-1}>
-    {getThumbnailAbbreviation(title)}
-  </Link>
-));
+>(({ title, to }, ref) => {
+  const { state, location } = useNavigation();
+  const shouldShowLoading = state !== "idle" && to === location?.pathname;
+  return (
+    <Link ref={ref} to={to} className={thumbnailStyle()} tabIndex={-1}>
+      {shouldShowLoading && <Loading delay={600} />}
```

Comment:
thumbnail link should have nothing to do with loading indication, they can be completely independent entities